test/unit/core/esriLayerUtils.spec.js

```diff
@@ -0,0 +1,184 @@
+        var ImageParameters;
+
+        beforeEach(function(){
+            ArcGISDynamicMapServiceLayer = jasmine.createSpy();
+            InfoTemplate = jasmine.createSpy();
+            ImageParameters = jasmine.createSpy();
+            // spyOn(ArcGISDynamicMapServiceLayer, 'setVisibleLayers');
+            esriLoader.require = function() {
+                return {
+                    then: function(callback) {
+                        callback([ArcGISDynamicMapServiceLayer, InfoTemplate, ImageParameters]);
+                    }
+                };
+            };
+        });
+
+        it('should pass just url to layer constructor if no options', function() {
+            esriLayerUtils.createDynamicMapServiceLayer('notARealUrl');
+            expect(ArcGISDynamicMapServiceLayer.calls.argsFor(0)).toEqual(['notARealUrl', undefined]);
+            expect(InfoTemplate.calls.count()).toEqual(0);
+            expect(ImageParameters.calls.count()).toEqual(0);
+        });
+
+        it('should pass url and options to layer constructor', function() {
+            var options = {
+                id: 'test'
+            };
+            esriLayerUtils.createDynamicMapServiceLayer('notARealUrl', options);
+            expect(ArcGISDynamicMapServiceLayer.calls.argsFor(0)).toEqual(['notARealUrl', options]);
+            expect(InfoTemplate.calls.count()).toEqual(0);
+            expect(ImageParameters.calls.count()).toEqual(0);
+        });
+
+        it('should normalize imageParameters option', function() {
+            var options = {
+                imageParameters: {
+                    format: 'jpeg'
+                }
+            };
+            esriLayerUtils.createDynamicMapServiceLayer('notARealUrl', options);
+            expect(ArcGISDynamicMapServiceLayer.calls.argsFor(0)).toEqual(['notARealUrl', options]);
+            expect(InfoTemplate.calls.count()).toEqual(0);
+            expect(ImageParameters.calls.count()).toEqual(1);
+        });
+
+        it('should normalize infoTemplates option', function() {
+            var options = {
+                infoTemplates: {
+                    1: {
+                        infoTemplate: {
+                            title: 'title1',
+                            content: 'content1'
+                        }
+                    },
+                    2: {
+                        infoTemplate: ['title2', 'content2']
+                    }
+                }
+            };
+            esriLayerUtils.createDynamicMapServiceLayer('notARealUrl', options);
+            expect(ArcGISDynamicMapServiceLayer.calls.argsFor(0)).toEqual(['notARealUrl', options]);
+            expect(InfoTemplate.calls.argsFor(0)).toEqual(['title1', 'content1']);
+            expect(InfoTemplate.calls.argsFor(1)).toEqual(['title2', 'content2']);
+            expect(ImageParameters.calls.count()).toEqual(0);
+        });
+
+        it('should set visibleLayers', function() {
+            // this feels dirty
```

Comment:
Yeah, my approach doesn't even work in all situations anyways...drives me crazy. My takeaway is that I do like what you ended up with and I want to try it out.